openshift/scripts/oc_deploy.sh

```diff
@@ -44,22 +44,46 @@ OC_PROCESS="oc -n ${PROJ_TOOLS} process -f ${PATH_DC} \
 OC_APPLY="oc -n ${PROJ_TARGET} apply -f -"
 [ "${APPLY}" ] || OC_APPLY="${OC_APPLY} --dry-run"
 
+# Cancel all previous deployments
+#
+OC_CANCEL_ALL_PREV_DEPLOY="oc -n ${PROJ_TARGET} rollout cancel dc/${NAME_OBJ} || true"
+
 # Deploy and follow the progress
 #
 OC_DEPLOY="oc -n ${PROJ_TARGET} rollout latest dc/${NAME_OBJ}"
 OC_LOG="oc -n ${PROJ_TARGET} logs -f --pod-running-timeout=1m dc/${NAME_OBJ}"
 if [ ! "${APPLY}" ]; then
-  OC_DEPLOY="${OC_DEPLOY} --dry-run"
+  OC_CANCEL_ALL_PREV_DEPLOY=""
+  OC_DEPLOY="${OC_DEPLOY} --dry-run || true" # in case there is no previous rollout
   OC_LOG=""
 fi
 
 # Execute commands
 #
 eval "${OC_PROCESS}"
 eval "${OC_PROCESS} | ${OC_APPLY}"
+eval "${OC_CANCEL_ALL_PREV_DEPLOY}"
+if [ "${APPLY}" ]; then # Check previous deployment statuses before moving onto new deploying
+  count=1
```

Comment:
Looping and sleeping is going to cause us some headaches - I don't have a better solution though.

Reply:
Yeah hopefully 10s timeout will help us ease some headaches